uber/site_sections/summary.py

```diff
@@ -30,6 +30,45 @@ def generate_staff_badges(start_badge, end_badge, out, session):
         badge_type_name='Staff').run(out, session)
 
 
+def volunteer_checklists(session):
+    attendees = session.query(Attendee) \
+        .filter(
+            Attendee.staffing == True,
+            Attendee.badge_status.in_([c.NEW_STATUS, c.COMPLETED_STATUS])) \
+        .order_by(Attendee.full_name, Attendee.id).all()
+
+    checklist_items = OrderedDict()
+    for item_template in c.VOLUNTEER_CHECKLIST:
+        item_name = os.path.splitext(os.path.basename(item_template))[0]
+        if item_name.endswith('_item'):
+            item_name = item_name[:-5]
+        item_name = item_name.replace('_', ' ').title()
+        checklist_items[item_name] = item_template
+
+    re_checkbox = re.compile(r'<img src="\.\./static/images/checkbox_.*?/>')
+    for attendee in attendees:
+        attendee.checklist_items = OrderedDict()
+        for item_name, item_template in checklist_items.items():
+            html = render(item_template, {'attendee': attendee}, encoding=None)
```

Comment:
Oh my goodness, are you... rendering the template to get the checklist completed status from it?

That is super clever, and will unexpectedly break if we ever decide to change how the templates are rendered. I wonder if there's a less brittle alternative.

Reply:
It's very clever, and very brittle, and also the only possible way to extract that information with the current system 😢